spinn_front_end_common/interface/front_end_common_interface_functions.py

```diff
@@ -429,10 +430,11 @@ def wait_for_cores_to_be_ready(
                                                            CPUState.C_MAIN)
         # check that everything has gone though c main to reach correct sync or
         # failing for some unknown reason
-        while processor_c_main != 0:
-            time.sleep(0.1)
-            processor_c_main = self._txrx.get_core_state_count(app_id,
-                                                               CPUState.C_MAIN)
+        while True:
```

Comment:
I very much disagree with using while True statements, and using break's to escape. They just look ugly and are rather prone to causing infinite loops. I try to only use them if theres no cleaner way to create the functionality. 
this all said, the version from master uses:

``` while processor_c_main != 0:
            time.sleep(0.1)
            processor_c_main = self._txrx.get_core_state_count(app_id,
                                                               CPUState.C_MAIN)```
and I think this is a much cleaner approach, therefore I wont be adding this bit in, unless theres some key reason for the while True functionality.

Please feel free to correct me if im wrong.